### DISCUSSION TITLE
Include and export 3D models with extension

### DISCUSSION THREAD

**Discussion by hannesspitzer** [foxglove/community#644] [Q&A]
Hi,
I'm working on a simple visualization that includes two custom 3D models (stl files). I have managed to get the models in the 3D panel by using the "ModelPrimitive" with an URL that points to a local directory. What I want to do is to export the 3D models together with the code so that they are part of the package. The next question would be how to reference the 3D models in the code correctly? There is also another possibility to use the "image data" (byte array) instead of the URL pointing to the image. 
Would be great if someone could help me solve this issue.

**Comment by jtbandes** [comment]
Yes, you can use the `data` and `media_type` fields of the ModelPrimitive message: https://foxglove.dev/docs/studio/messages/model-primitive

For STL data, media_type can be "`model/stl`".
